software/ds4_control.py

```diff
@@ -187,62 +193,71 @@ def on_square_press(self):
                 with open(self.data_file, "a") as f:
                     f.write(self.last_throw_data)
                 self.last_throw_data = str()
-                print("Wrote speeds to file.")
+                self.robot_data.logger.log.info("Wrote speeds to file.")
             else:
-                print("Nothing to write!")
+                self.robot_data.logger.log.warning("Nothing to write!")
 
     def on_circle_press(self):
         """Attempt to throw the ball"""
         if self.robot_data.current_state == State.RemoteControl:
```

Comment:
I guess the remote controller manipulates the robots state here? Usually not too good for multiple places to modify the state since it is harder to keep up with what is responsible for what. Cant a button press toggle a state inside the main logic and it does its behavior there?

Reply:
It doesn't touch the robot's state here. This is based on a modified version of the throw state from main logic, but with extra logging, so we can save thrower data more easily when trying to calibrate it.